tox.ini

```diff
@@ -56,9 +55,18 @@ commands = pre-commit run --config .pre-commit-config.yaml.failing --all-files {
 [testenv:setup-check]
 deps =
     twine
-extras =
-    docs
```

Comment:
I think we should check docs separately from the test suite because this way different types of CI failures are better discoverable and recognizable

Reply:
Yes, maybe. I do want to keep things simple so they scale from tiny projects to larger ones. The most important part of this change is that it eliminates support on invoking 'setup.py' (such that only pip does that and so it supplies all dependencies).

Reply:
I think some of pip maintainers want `--only-extras` feature for this case... So it still needs to be pulled in from an extra